working/augmentation-libraries/feature-specification.md

```diff
@@ -85,58 +85,57 @@ imports and prefixes of their parent (part or library) file.
 
 Augmentation declarations interact with part files mainly in restrictions on
 where an augmenting declaration may occur relative to the declaration it
-augments, as describe below.
+augments, as described below.
 
 For this, we define the following relations on *declarations* based on the
 relations between *files* of a library.
 
 We say that a syntactic declaration *occurs in* a Dart file if the
-declaration’s source code occurs in that Dart file.
+declaration's source code occurs in that Dart file.
 
 We then say that a Dart file *contains* a declaration if the declaration occurs
-in the file itself, or if any of the files included by the Dart file contains
-the declaration. _That is, if the declaration occurs in a file in the sub-tree
+in the file itself, or if any of the files included by the Dart file contain
+the declaration. _That is, if the declaration occurs in a file in the subtree
 of that Dart file._
 
-We then define a partial and a complete *ordering* of declarations of a library
-as follows:
+We then define two *orderings* of declarations in a library, one partial and one
+complete, as follows:
 
 We define a partial ordering on syntactic declarations of a library,
 *is above*, such that a syntactic declaration *A* is *above* a syntactic
 declaration *B* if and only if:
 
-*   *A* and *B* occur in the same file, and the start of the *A* declaration is
-syntactically before the start of the *B* declaration, in source order, or
-*   A file included by the file containing *A* contains *B*.
+*   *A* and *B* occur in the same file, and the start of *A* is syntactically
+    before the start of *B*, in source order, or
+*   The file containing *A* includes the file that contains *B*.
```

Comment:
Just double checking the earlier text in the current version of the feature specification, I can see that 'contains' is transitive (meaning "occurs in a file which is in the subtree"), and 'includes' is the simple (and non-transitive) relationship declared by `part ...;`.

First of all, this means that "the file containing *A*" is ambiguous, and we'd probably want to say something like "the deepest file containing *A*". This is a file _F_ that contains *A* such that there is no file _F1_ which is included by _F_ where _F1_ contains *A* as well.

We could also talk about "contains *A*" meaning "the declaration of *A* is located here" and "covers *A*" meaning "contains *A* or includes a file that contains *A*", but that's a bigger change because we'd then need to change 'contains' to 'covers' almost everywhere.

In any case, we could say something like this: "The deepest file containing *A* includes a file that contains *B*."

(With "covers" that would be: "The file that contains *A* includes a file that covers *B*".)

Reply:
Or just use the same phrase used to define "contains", "occurs":

> * The file where *A* occurs includes the file where *B* occurs.

Reply:
Good points! I like Lasse's phrasing. I'd like to avoid introducing any new terminology like "covers" if possible.